### PR TITLE
Feature/ajw duplication fix cleanup

### PR DESCRIPTION
**Pull request by adamjohnwright**

### CODE REVIEW

**Comment by adamjohnwright** [comment]
First if you could check to see if thes changes are all logical. I know there are lot. I still have to figure out if it will work with scheduling many workflow versions. I am not sure what else is missing but a couple things I am sure. I made it so that all flags can be specified in the decider.ini file. In fact that would be prefered - I probably have to check to make sure the command line flags fully line up with the code fully. 

Once we merge it we can make a realeas branch and test it on AWS, as you suggesed. 

Please take a careful look at the code. I think there are still things that aren't working with the code. I don't know if I made it so that something is not working that was. I don't think so but it is possilbe. I have tested it on ETRI and it appeared as if it is creating the correct ini files. 

Let me know if you see anything that should be fixed right away and I will get on it. 

Cheers,
AJW

**Comment by briandoconnor** [comment]
Would be really nice to have the following be filled in in the workflow template from the settings file rather than hardcoded:
- cleanup=
- study-refname-override=
- analysis-center-override=
- seqware-output-lines-number=

That way you don't really have site-specific workflow.ini, you can just use the same template and make your changes in one decider config. 

On a related note, we could then have a template per profile (8 cores, 32G, 16 cores, 64G, etc) rather than site-specific.  This would help us to converge on settings that work across sites that share the same profile.

Finally, it would be very helpful to point the decider at a specific workflow template .ini to use rather than it just guessing based on the version of the workflow.

**Comment by adamjohnwright** [comment]
I will add these in tomorrow. I will let you know how it goes

File: lib/GNOS/SampleInformation.pm [hunk 1]
```diff
@@ -98,12 +101,12 @@ sub get {
...
 
         while ($status == 0 and $attempts < 10) {
-            $status = $self->download_analysis($analysis_full_url, $analysis_xml_path, $use_cached_xml);
+            $status = $self->download_analysis($analysis_full_url, $analysis_xml_path, $use_cached_analysis);
             $attempts++;
         }
 
         if (not -e $analysis_xml_path or not eval {$xs->XMLin($analysis_xml_path); } ) {
-           say $parse_log "skipping $analysis_id - no xml file available: $analysis_xml_path";
+           say $parse_log "SKIPPING: $analysis_id - no xml file available: $analysis_xml_path";
```
Comment (briandoconnor) [inline lib/GNOS/SampleInformation.pm]: Isn't this fatal?  Shouldn't this cause the program to die?

**Comment by briandoconnor** [comment]
The local cached file is completely broken.

Before:

692813f3-131f-4b2d-a37a-a3d2a7cc4c6a,f27252ce-2638-49e7-869e-869954d0bc06       2015-01-26T05:19:58.626Z                running PACA-AU ICGC_0124

After:

1c9fbe26-1447-4335-aaab-06d6bb3e2741    2015-01-26T05:19:58.626Z        unknown running

File: lib/GNOS/SampleInformation.pm (hunk 1 above)
Comment (adamjohnwright) [inline lib/GNOS/SampleInformation.pm]: I hav echanged it for a few that are actually dieing to say so! ... Including this one.

**Pull request merged**